File: 0728-self-dividing-numbers/0728-self-dividing-numbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    let selfDividingNums = [];
    for (i = left; i <= right; i++) {
        if (eachDigit(i)) {
            selfDividingNums.push(i);
        }
    }
    return selfDividingNums;

    function eachDigit(num) {
        if (num < 10) {
            return true;
        }
        let temp = num;
        while (temp > 0) {
            let cur = temp % 10;
            if (cur === 0 || num % cur !== 0) {
                return false;
            }
            temp = Math.floor(temp / 10);
        }
        return true;
    }
};

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        selfDividingNums = []

        while left <= right:
            is_self_dividing = True

            for d in str(left):
                if int(d) == 0 or int(left) % int(d) != 0:
                    is_self_dividing = False
            
            if is_self_dividing:
                selfDividingNums.append(int(left))
            
            left += 1
        
        return selfDividingNums